kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/DisbursementVoucherNonResidentAlienInformationValidation.java

```diff
@@ -0,0 +1,341 @@
+                return false;
+            }
+            // Exempt Under Other Code should be checked; this will ensure the other 3 check boxes not checked due to mutual exclusiveness
+            if( !nonResidentAlienTax.isTaxOtherExemptIndicator() ) {
+                errors.putErrorWithoutFullErrorPath("DVNRATaxErrors", KFSKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_SHOULD_BE_SELECTED, "USAID Per Diem", "Exempt Under Other Code");
+                return false;
+            }
+            // Special W-4 Amount shall have no value 
+            if( nonResidentAlienTax.getTaxSpecialW4Amount() != null ) {
+                errors.putErrorWithoutFullErrorPath("DVNRATaxErrors", KFSKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_CANNOT_HAVE_VALUE, "USAID Per Diem", "Special W-4 Amount");
+                return false;
+            }
+        }
+
+        // if Special W-4 Amount is entered
+        if( nonResidentAlienTax.getTaxSpecialW4Amount() != null ) {
+            // Conditions to be met for "Special W-4 Amount" error to be generated
+            // income class code should be fellowship
+            if( !nonResidentAlienTax.getIncomeClassCode().equals(NRA_TAX_INCOME_CLASS_FELLOWSHIP) ) {
+                errors.putErrorWithoutFullErrorPath("DVNRATaxErrors", KFSKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_SHOULD_BE_SELECTED, "Special W-4 Amount", "Income Class Code : Fellowship" );
+                return false;
+            }
+            // Federal and State tax rate should be zero.
+            if( !(nonResidentAlienTax.getStateIncomeTaxPercent().equals(KualiDecimal.ZERO)) || !(nonResidentAlienTax.getFederalIncomeTaxPercent().equals(KualiDecimal.ZERO)) ) {
+                errors.putErrorWithoutFullErrorPath("DVNRATaxErrors", KFSKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_FEDERAL_AND_STATE_TAXES_SHOULD_BE_ZERO , "Special W-4 Amount");
+                return false;
+            }
+            // Exempt Under Other Code should be checked; this will ensure the other 3 check boxes not checked due to mutual exclusiveness
+            if( !nonResidentAlienTax.isTaxOtherExemptIndicator() ) {
+                errors.putErrorWithoutFullErrorPath("DVNRATaxErrors", KFSKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_SHOULD_BE_SELECTED, "Special W-4 Amount", "Exempt Under Other Code");
+                return false;
+            }
+            // USAID Per Diem should not be checked (mutual exclusive checking on USAID Per Diem here is optional since this is also ensured by validation on Special W-4 Amount above   
+            if( ( nonResidentAlienTax.isTaxUSAIDPerDiemIndicator() ) ) {
+                errors.putErrorWithoutFullErrorPath("DVNRATaxErrors", KFSKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_CANNOT_BE_SELECTED, "Special W-4 Amount", "USAID Per Diem");
+                return false;
+            }
+        }
+
+        // if NQI Id is entered
+        if( nonResidentAlienTax.getTaxNQIId() != null) {

```

Comment:
Is this similar to the ICC stubbed if-blocks? That is, is this a placeholder for another step to come? I'm totally fine with it if so, just wanted to be sure.